src/main/java/de/tum/in/www1/artemis/service/programming/ProgrammingExerciseFeedbackCreationService.java

```diff
@@ -45,8 +45,24 @@ public class ProgrammingExerciseFeedbackCreationService {
 
     private static final Predicate<String> IS_PYTHON_EXCEPTION_LINE = line -> line.startsWith(PYTHON_EXCEPTION_LINE_PREFIX);
 
-    private static final List<String> TIMEOUT_EXCEPTIONS = Arrays.asList("org.junit.runners.model.TestTimedOutException", "java.util.concurrent.TimeoutException",
-            "org.awaitility.core.ConditionTimeoutException", "Timed?OutException");
+    private static final String TIMEOUT_EXCEPTIONS_REGEX = String.join("|", "org.junit.runners.model.TestTimedOutException", "java.util.concurrent.TimeoutException",
+            "org.awaitility.core.ConditionTimeoutException", "TimedOutException");
```

Comment:
```suggestion
            "org.awaitility.core.ConditionTimeoutException", "Timed?OutException");
```
Difference compared to old variant. Does that matter? I assume `TimeOutException`s might exist somewhere.